module-services/service-desktop/parser/ParserFSM.cpp

```diff
@@ -7,20 +7,32 @@
 #include <memory>
 #include <string>
 
+#include <Timers/TimerFactory.hpp>
 #include "MessageHandler.hpp"
 #include <endpoints/EndpointFactory.hpp>
 #include <endpoints/message/Common.hpp>
 
+namespace
+{
+    constexpr auto parserTimerDelayMs = std::chrono::milliseconds{1000 * 5};
+    constexpr auto parserTimerName    = "parserTimer";
+} // namespace
+
 namespace sdesktop::endpoints
 {
 
-    StateMachine::StateMachine(sys::Service *OwnerService) : OwnerServicePtr(OwnerService)
+    StateMachine::StateMachine(sys::Service *OwnerService)
+        : OwnerServicePtr(OwnerService),
+          parserTimer{sys::TimerFactory::createPeriodicTimer(
```

Comment:
Probably you should use SingleShotTimer instead of periodic one.

Reply:
Right, this makes more sense.